test/utils.cpp

```diff
@@ -608,32 +591,80 @@ TensorView* biasEpilogue(TensorView* tensor, TensorView* bias) {
       bias->dtype(),
       ", tensor: ",
       tensor->dtype());
-  const auto concrete = TensorDomain::noReductions(
-      TensorDomain::noBroadcasts(tensor->getLeafDomain()));
-  TORCH_CHECK(
-      concrete.size() == 2,
-      "Only tensors with two concrete domains have support for bias epilogue enabled, got ",
-      concrete.size());
   TORCH_CHECK(
       tensor->nDims() >= 2,
       "Tensors to have bias applied needs to have 2 or more domains, got ",
       tensor->nDims());
-  TORCH_CHECK(
-      bias->nDims() == 1,
-      "bias vector must have one dimension, got",
-      bias->nDims());
+
+  const auto concrete = TensorDomain::noReductions(
+      TensorDomain::noBroadcasts(tensor->getLeafDomain()));
 
   TensorView *biasb = nullptr, *biased = nullptr;
-  biasb = broadcast(bias, {false, true});
+
+  switch (concrete.size()) {
+    case 2:
+      // regular matmul (non-strided batch gemm)
+      TORCH_CHECK(
+          bias->nDims() == 1,
+          "bias vector must have one domain, got",
+          bias->nDims());
+      biasb = broadcast(bias, {false, true});
+      break;
+    case 3:
+      // strided batch gemm case
+      TORCH_CHECK(
+          (bias->nDims() == 2 || bias->nDims() == 1),
+          "bias vector must have one (single bias for batch) "
+          "or two (bias for each batch entries)), got",
+          bias->nDims());
+      biasb = broadcast(bias, {false, false, true});
+      break;
+    default:
+      TORCH_CHECK(
+          false,
+          "Only tensors with two (matmul) or three (strided batch matmul) "
+          "concrete domains have support for bias epilogue enabled, got ",
+          concrete.size());
+  }
+
   biased = add(tensor, biasb);
   return biased;
 }
 
 at::Tensor atBiasEpilogue(const at::Tensor& tensor, const at::Tensor& bias) {
-  TORCH_CHECK(bias.dim() == 1, "Bias must be a vector");
-  TORCH_CHECK(
-      tensor.dim() == 2, "Only 2d tensors have support for bias epilogue");
-  const int64_t rows = bias.size(0);
+  switch (tensor.dim()) {
+    case 2:
+      TORCH_CHECK(
+          bias.dim() == 1,
+          "For single matmul problem bias must be a vector, got ",
+          bias.dim());
+      TORCH_CHECK(
+          tensor.dim() == 2,
+          "For single matmul problem only 2d tensors have"
+          " bias epilogue support, got ",
+          tensor.dim());
```

Comment:
This check is redundant, similar below.

Reply:
You are right, this is a leftover from earlier version.

Will be fixed with the follow up commit. Thanks!